utils/src/main/kotlin/Utils.kt

```diff
@@ -379,7 +379,13 @@ fun JsonNode?.asTextOrEmpty(): String = if (this != null) this.asText() else ""
  */
 fun String.fileSystemEncode() =
         // URLEncoder does not encode "." and "*", so do that manually.
-        java.net.URLEncoder.encode(this, "UTF-8").replace("*", "%2A").replace(".", "%2E")
+        java.net.URLEncoder.encode(this, "UTF-8").replace("*", "%2A").let {
+            when {
+                it.matches(Regex("\\..+")) -> it.replace(".", "%2E")
```

Comment:
> [...] or if the string consists only of dots.

Shouldn't this say "or if the string contains at least two consecutive dots"? Like, we *do* want to encode strings like "foo..bar", right?

Anyway, the Regex does yet something different as there is a second (un-escaped) `.`. I believe we want something like this:

    java.net.URLEncoder.encode(this, "UTF-8").replace("*", "%2A").replace(Regex("(^\\.|\\.\\.+)")) {
        it.value.replace(".", "%2E")
    }

Reply:
Not sure, doe we? To me "foo..bar" would be fine, only ".", "..", and ".something" cause issues, right?

Reply:
Actually I think only replace the dot if it is the first character should be enough to get valid filenames. Would you agree with these specs?

```
. -> %2E
.. -> %2E.
... -> %2E..
test -> test
test.test -> test.test
test..test -> test..test
.test.test -> %2Etest.test
..test -> %2E.test
```

Reply:
Done.